mne/io/tests/test_raw.py

```diff
@@ -281,3 +281,46 @@ def test_get_data_reject():
         assert log.getvalue().strip() == msg
     assert data.shape == (len(ch_names), 2560)  # shape doesn't change
     assert np.isnan(data).sum() == 3072  # but NaNs are introduced instead
+
+
+def test_5839():
+    """Test concatenating raw objects with annotations."""
+    # Global Time 0         1         2         3         4
+    #             .
+    #      raw_A  |---------XXXXXXXXXX
+    #      annot  |--------------AA
+    #    latency  .         0    0    1    1    2    2    3
+    #             .              5    0    5    0    5    0
+    #
+    #      raw_B  .                   |---------YYYYYYYYYY
+    #      annot  .                   |--------------AA
+    #    latency  .                             0         1
+    #             .                                  5    0
+    #             .
+    #     output  |---------XXXXXXXXXXYYYYYYYYYY
+    #      annot  |--------------AA---|----AA
+    #    latency  .         0    0    1    1    2    2    3
+    #             .              5    0    5    0    5    0
+    #
+    EXPECTED_ONSET = [1.5, 2.5, 2., 2.]
+    EXPECTED_DURATION = [0.2, 0.2, 0., 0.]
+    EXPECTED_DESCRIPTION = ['dummy', 'dummy', 'BAD boundary', 'EDGE boundary']
```

Comment:
LGTM but can someone remind me why we have both 'BAD boundary' and 'EDGE boundary'?

Reply:
To behave during filtering like EEGLAB (only split up filtering based on EDGE events, ignore other event types), which seemed like a reasonable thing to do 

https://github.com/mne-tools/mne-python/issues/4353#issuecomment-311387886

Feel free to add this URL in a comment somewhere you would have seen it